golang/update.sh

```diff
@@ -3,23 +3,48 @@
 set -euExo pipefail
 shopt -s inherit_errexit
 
-source "$( dirname "${BASH_SOURCE[0]}" )/lib.sh"
-
-# Fetch the latest versions
-version_info="$( curl --fail --retry 5 --retry-all-errors -L  https://go.dev/dl/?mode=json )"
+function get-url {
+  echo "https://go.dev/dl/go${1}.linux-${2}.tar.gz"
+}
 
 function get-latest-version {
-  echo "${version_info}" | jq -r '.[] | select( .version | test("'"${1}"'") ) | .version | sub("^go"; "")'
+  echo "${1}" | jq -r '.[] | select( .version | test("'"${2}"'") ) | .version | sub("^go"; "")'
+}
+
+function dump-versions {
+  output_file=${1}
+  go_version=${2}
+  golangci_lint_version=${3}
+  cat <<EOF > ${output_file}
+# This file is auto-generated by golang/update.sh
+GOLANG_VERSION="${go_version}"
+GOLANGCI_LINT_VERSION="${golangci_lint_version}"
+EOF
 }
 
 for version_regex in 'go1\\.23($|\\.[0-9]+)' 'go1\\.24($|\\.[0-9]+)'; do
-  version=$( get-latest-version "${version_regex}" )
+  # Resolve golang version.
+  version_info="$( curl --fail --retry 5 --retry-all-errors -L  https://go.dev/dl/?mode=json )"
+  version=$( get-latest-version "${version_info}" "${version_regex}" )
   short_version="${version%.*}"
-  find "${short_version}" -name '*.sum' -delete
+
+  # Resolve golangci-lint version.
+  golangci_lint_version="$( curl --fail --retry 5 --retry-all-errors -L https://api.github.com/repos/golangci/golangci-lint/releases/latest | jq -r '.tag_name' )"
```

Comment:
why tie this together with the golang version update? Is there any relation between the two?

Reply:
Tied them because the images are grouped by Golang version. For each Golang version, we have to keep the checksum of golangci-lint that it installs. It also makes sense to have some automation to automatically bump the golangci-lint version (to avoid manual version bump and checksum save), and this covers that. If you think this should be somehow separated, I'm open to suggestions.

In general, golangci-lint is compatible with Golang versions lower or equal to the one it was built with. It means that once a breaking change to some Golang API is introduced, golangci-lint has to adapt.